lib/SlackBot.js

```diff
@@ -0,0 +1,331 @@
+          var issue = listOfIssues[j];
+
+          if (defined(issue.pull_request)) {
+            var endTime = new Date(issue.closed_at);
+            var startTime = new Date(issue.created_at);
+            var daysDifference = (endTime - startTime) / (24 * 60 * 60 * 1000);
+            mergeTimes.push(daysDifference);
+
+            if (!defined(longestPullRequest) || daysDifference > longestMergeTime) {
+              longestPullRequest = issue;
+              longestMergeTime = daysDifference;
+            }
+
+          }
+        }
+      }
+    })
+    .then(function() {
+      var averageMergeTime = average(mergeTimes).toFixed(1);
+      longestMergeTime = longestMergeTime.toFixed(1);
+
+      // Compute the standard deviation. If the longest merge time is more than 3 times that, report is as unusual.
+      var unusuallyLongPRMessage = '';
+      var deviation = standardDeviation(mergeTimes);
+
+      if (Math.abs(longestMergeTime - averageMergeTime) >= deviation * 2) {
+        unusuallyLongPRMessage = 'This PR (' + longestPullRequest.html_url + ') took an unusually long amount of time to get merged (' + longestMergeTime + ' days).';
+      }
+
+      var greetings = ['Happy Friday everyone!', 'Can you believe Friday is already here?', 'I hope you all had awesome week!',
+                      'I skipped breakfast, so I hope Gary baked something good today...', 'Good morning everyone!'];
+
+      var greeting = greetings[Math.floor(Math.random() * greetings.length)];
+      if (defined(disableRandomGreeting)) {
+        greeting = greetings[0];
+      }
+
+      var templateName = 'weeklyStats';
+      var template = fs.readFileSync(path.join(__dirname, 'templates', templateName + '.hbs')).toString();
+      var messageText = handlebars.compile(template)({
+        greeting : greeting,
+        averageMergeTime : averageMergeTime,
+        numberOfPullRequests : mergeTimes.length,
+        unusuallyLongPRMessage : unusuallyLongPRMessage
+      });
+
+      that.postMessage(that._channelIDs['general'], messageText);
+    });
+};
+
+SlackBot._sendReleaseReminders = function() {
+  // Update config before we post any reminders,
+  // that way, we can update the config and the bot will always have the latest copy.
+  var that = this;
+  return this._getConfig(this._configUrl)
+  .then(function(releaseSchedule) {
+    for (var user in releaseSchedule) {
+      if (releaseSchedule.hasOwnProperty(user)) {
+        var days = Math.floor(daysSince(releaseSchedule[user]));
+        var templateName;
+
+        if (days === -14) {
+          templateName = 'releaseReminderEarly';
+        } else if (days === -7) {
+          templateName = 'releaseReminder';
+        } else if (days === -1) {
+          templateName = 'releaseReminderLate';
+        }
+
+        if (defined(templateName)) {
+          var userObject = that._userData[that._userIDs[user]];
+          var firstName = userObject.real_name.split(' ')[0];
```

Comment:
This one is tricky. There is a returned `first_name` but that may not always be set (for example, Sarah has no first_name set). I changed it to just use `display_name` which I think is a safe bet since this is most often how your name appears in Slack.